PLC Development/DrivePLC/POUs/FB_AxisDrive.TcPOU

```diff
@@ -0,0 +1,868 @@
+		
+		bDone:=MCWriteBoolParameter.Done;
+		AxisController.bBusy:=MCWriteBoolParameter.Busy;
+		
+		IF MCWriteBoolParameter.Error THEN
+			AxisController.bError:= TRUE;
+			AxisController.iErrorID:=MCWriteBoolParameter.ErrorID;
+		END_IF			
+		IF NOT MCWriteBoolParameter.Busy THEN
+			eCommandLatched:=eMcCommand.Idle;
+		END_IF	
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	eMcCommand.SetPosition:	
+		MCSetPosition.Execute:=AxisController.bExecute;
+		MCSetPosition.Position:=AxisController.fPosition;
+		
+		AxisController.bBusy:=MCSetPosition.Busy;
+		bDone:=MCSetPosition.Done;
+		
+		IF MCSetPosition.Error THEN
+			AxisController.bError:=TRUE;
+			AxisController.iErrorID:= MCSetPosition.ErrorID;
+		END_IF
+		IF NOT MCSetPosition.Busy THEN
+			eCommandLatched:=eMcCommand.Idle;
+		END_IF
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	eMcCommand.MoveAbsolute:	
+		MCMoveAbsolute.Execute:=AxisController.bExecute;
+		MCMoveAbsolute.Position:=AxisController.fPosition;
+		MCMoveAbsolute.Velocity:=AxisController.fVelocity;
+		MCMoveAbsolute.Acceleration:=AxisController.fAcceleration;
+		MCMoveAbsolute.Deceleration:=AxisController.fDeceleration;
+		MCMoveAbsolute.Jerk:=0;
+		MCMoveAbsolute.BufferMode:=MC_BufferMode.MC_Buffered;
+		
+		bDone:=MCMoveAbsolute.Done;
+		AxisController.bBusy:=MCMoveAbsolute.Busy;
+		
+		IF MCMoveAbsolute.Error THEN
+			AxisController.bError:=TRUE;
+			AxisController.iErrorID:=MCMoveAbsolute.ErrorID;
+		END_IF
+		IF NOT MCMoveAbsolute.Busy THEN
+			eCommandLatched:=eMcCommand.Idle;
+		END_IF
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	eMcCommand.MoveRelative:		
+		MCMoveRelative.Execute:=AxisController.bExecute;
+		MCMoveRelative.Distance:=AxisController.fDistance;
+		MCMoveRelative.Velocity:=AxisController.fVelocity;
+		MCMoveRelative.Acceleration:=AxisController.fAcceleration;
+		MCMoveRelative.Deceleration:=AxisController.fDeceleration;
+		MCMoveRelative.Jerk:=0;
+		MCMoveRelative.BufferMode:=MC_BufferMode.MC_Buffered;
+		
+		bDone:=MCMoveRelative.Done;
+		AxisController.bBusy:=MCMoveRelative.Busy;
+		
+		IF MCMoveRelative.Error THEN
+			AxisController.bError:=TRUE;
+			AxisController.iErrorID:=MCMoveRelative.ErrorID;
+		END_IF
+		IF NOT MCMoveRelative.Busy THEN
+			eCommandLatched:=eMcCommand.Idle;
+		END_IF		
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	eMcCommand.GearIn:
+		//Dependent on +ve or -ve gear ratio determines whether to link Fw to Fw and Bw to Bw or Fw to Bw and Bw to Fw
+		IF AxisController.fGearRatioNum>0 THEN
+			Limits.coupledFwLimit[GVL.iAxis][1].eLimitType:=eLimitCoupling.FwSw;
+			Limits.coupledFwLimit[GVL.iAxis][1].iMasterAxis:=AxisController.masterAxis;
```

Comment:
You're calling this same line 4 times in these if statements. I think you can do it in 1.